File: go/leap/leap.go
package leap

const testVersion = 2

func divisibleBy(n int, year int) bool {
	return year%n == 0
}

func IsLeapYear(year int) bool {
	return divisibleBy(400, year) ||
		(divisibleBy(4, year) && !divisibleBy(100, year))
}

File: dart/leap/lib/leap.dart
class Year {
  int year;
  Year(this.year);

  bool divisibleBy(int n) {
    return year % n == 0;
  }

  bool isLeap() {
    return divisibleBy(400) || (divisibleBy(4) && !divisibleBy(100));
  }
}
